### PR TITLE
App launch timeout config

### PR DESCRIPTION
**Pull request by skywilliamsonl**
Summary: Make session server app timeout configurable, where default is no timeout.
Type: Feature
Test Plan: Fullstack application manual testing
Dependencies and Impacts: None
Jira: RIALTO-35

### CODE REVIEW

**Review by skywojciechowskim — APPROVED** [review]

File: serverManager/common/source/SessionServerApp.cpp [hunk 1]
```diff
@@ -54,6 +53,25 @@ std::string getSessionServerPath()
     }
     return "/usr/bin/RialtoServer";
 }
+
+std::chrono::milliseconds getStartupTimeout()
+{
+    const char *customTimeout = getenv("RIALTO_SESSION_SERVER_TIMEOUT_MS");
+    std::chrono::milliseconds timeout{0};
```
Comment (aczs) [inline serverManager/common/source/SessionServerApp.cpp:60]: Do we want timeout to be disabled by default?
In my opinion, timeout should be disabled only for testing purposes, so we will always have to set it in production code.

File: serverManager/common/source/SessionServerApp.cpp [hunk 2]
```diff
@@ -54,6 +53,25 @@ std::string getSessionServerPath()
     }
     return "/usr/bin/RialtoServer";
 }
+
+std::chrono::milliseconds getStartupTimeout()
+{
+    const char *customTimeout = getenv("RIALTO_SESSION_SERVER_TIMEOUT_MS");
```
[the remark this answers is not in the document]
Comment (skywilliamsonl) [reply serverManager/common/source/SessionServerApp.cpp]: Yeah, agree its more explicit.

File: serverManager/common/source/SessionServerApp.cpp [hunk 3]
```diff
@@ -54,6 +53,25 @@ std::string getSessionServerPath()
...
+    std::chrono::milliseconds timeout{0};
+    if (customTimeout)
+    {
+        try
+        {
+            timeout = std::chrono::milliseconds(std::stoull(customTimeout));
+            RIALTO_SERVER_MANAGER_LOG_WARN("Using custom SessionServer startup timeout: %sms", customTimeout);
```
[the remark this answers is not in the document]
Comment (skywilliamsonl) [reply serverManager/common/source/SessionServerApp.cpp]: Changed to info.

File: serverManager/common/source/SessionServerApp.cpp [hunk 4]
```diff
@@ -54,6 +53,25 @@ std::string getSessionServerPath()
...
+        }
+        catch (const std::exception &e)
+        {
+            RIALTO_SERVER_MANAGER_LOG_ERROR("Custom SessionServer startup timeout invalid, ignoring: %s", customTimeout);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (skywilliamsonl) [reply serverManager/common/source/SessionServerApp.cpp:72]: Done.

**Review by skypetts — APPROVED** [review]

**Review by skywojciechowskim — APPROVED** [review]

**Pull request merged**